app/upgrades/v9/upgrades.go

```diff
@@ -20,13 +20,21 @@ import (
 	"github.com/osmosis-labs/osmosis/v7/app/keepers"
 )
 
+const preUpgradeAppVersion = 8
+
 func CreateUpgradeHandler(
 	mm *module.Manager,
 	configurator module.Configurator,
 	keepers *keepers.AppKeepers,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
 		ExecuteProp214(ctx, keepers.GAMMKeeper)
+    
+    // We set the app version to pre-upgrade because it will be incremented by one
```

Comment:
```suggestion
        // We set the app version to pre-upgrade because it will be incremented by one
```

(I messed this up in merge conflict resolution)